apps/src/dance/ai/DanceAiModal.tsx

```diff
@@ -451,6 +462,31 @@ const DanceAiModal: React.FunctionComponent = () => {
 
   const labels = getLabels();
 
+  const calculateMinMax = () => {
+    return Array.from(Array(BAD_GENERATED_RESULTS_COUNT + 1).keys()).reduce(
+      (accumulator: MinMax, currentValue: number) => {
+        const scores = getScores(currentValue);
+        const min = Math.min(...scores);
+        const max = Math.max(...scores);
+        if (min < accumulator.min) {
+          accumulator.min = min;
+        }
+        if (max > accumulator.max) {
+          accumulator.max = max;
+        }
+        return accumulator;
+      },
+      {min: Infinity, max: 0}
+    );
+  };
+
+  if (
+    generatedEffects.current.goodEffect &&
+    generatedEffects.current.badEffects.length === BAD_GENERATED_RESULTS_COUNT
+  ) {
+    minMaxAssociations.current = calculateMinMax();
+  }
```

Comment:
It would be handy to just call this when we set the good/bad effects, but understood that this code depends on other state being set (e.g. the inputs) so this seems fine for now.